### DISCUSSION TITLE
✨ June 5, 2023 Updates

### DISCUSSION THREAD

**Discussion by LarsKemmann** [CareTogether/CareTogetherCMS#656] [Announcements]
By now, some of you may have noticed the new experimental feature that was released over the weekend:

### 🎉 Introducing the Calendar Dashboard! 🚀
This is probably the most-requested feature from you all, and I'm super excited to show you the first draft and get your advice on how to make it better!

![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/60013e76-13d3-4dea-bd4d-5b438c3209ba)

CareTogether has had a (mostly useless) "dashboard" page for a while, but now when you browse to the Dashboard section you'll see an at-a-glance view of the current arrangements for your location! This includes the planned and actual durations, planned and actual childcare, and completed/upcoming/past-due requirements. Since this can quickly become overwhelming if there are many arrangements going on at the same time, there's also a filter option to focus on just the information you care about:

![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/32e01592-33f5-46c9-a342-a3c63b510875)

This calendar view is optimized for desktop viewing, but there is also a list view that is slightly more mobile-friendly:

![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/5a4adab3-575d-4ff6-a00e-6ea616c900a6)

This is just the first draft—I have lots of additional improvements planned for the dashboard!! They may have to wait a little bit because, you know, 👶🏻... 🥰

### ✨ Other Improvements 🥳

Arrangements can now be assigned a **reason for request** to help with reporting as well as classifying arrangements by the severity of the need.

![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/81de2ed6-499c-4411-b089-f58b77cc8f66)

The Referrals screen now has a filter to let you switch between seeing all​ partnering families (and all their arrangements, past and present), and seeing only active arrangements. This filter will remember what you set it to, so for those of you who mostly supervise ongoing arrangements you'll be able to see just what you need to pay attention to whenever you go to the Referrals screen!

![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/2da7af0f-8968-4e8d-91be-2a0332278035)

Upcoming but not past-due requirements will no longer show red X's on arrangement cards! Instead, you'll see a nice friendly "info" icon, so for those of you who (like me) really thrive on keeping things organized, hopefully CareTogether will induce a little less anxiety from now on. 😄 Special thanks to new contributor @jaysengardner for building this! 🙌🏻 #646 

![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/37fc72ba-e66e-4c80-9ed9-c8a29bbf82b8)

### 🐛 Bugs Fixed

Self-service password reset (for anyone not using a Microsoft 365 staff account) now actually works!

![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/b46e87aa-b6f0-4e94-8b3b-641a5e50c36f)
![image](https://github.com/CareTogether/CareTogetherCMS/assets/5834429/348144f7-ad37-4933-89f1-4156306686d5)

A critical bug that would crash the system if you tried ending an arrangement before its start date has been fixed!! Special thanks to new contributor @Ad-Mortem for building this! 🙌🏻 #524 

Several bugs in the login logic for CareTogether have been fixed, which means 1) you should have a much more seamless time with signing in and 2) in the process it's possible that one or two new bugs might have been introduced 🐜 — so please keep an eye out for any login/session-timeout issues and let me know how it's going!

~Lars

**Comment by bsdoolittle** [comment]
Love these updates...especially those related to reporting.  Great job Lars!
